library/init/coe.lean

```diff
@@ -41,8 +41,8 @@ class has_coe (a : Sort u) (b : Sort v) :=
 class has_coe_t (a : Sort u) (b : Sort v) :=
 (coe : a → b)
 
-class has_coe_to_fun (a : Sort u) : Sort (max u (v+1)) :=
-(F : a → Sort v) (coe : Π x, F x)
+class has_coe_to_fun (a : Sort u) : Sort (max u (v+1) (w+1)) :=
+(domain : a → Sort v) (codomain : Π {x}, domain x → Sort w) (coe : Π x, Π (m : domain x), codomain m)
```

Comment:
Issue 1: I can't make `domain` and `codomain` implicit, despite them being inferrable much of the time, because then I can't nicely specify them when they *aren't* inferrable. Why can't we specify implicit fields?